libraries/Microsoft.Bot.Builder.Ai/LocaleConverter.cs

```diff
@@ -11,41 +11,63 @@
 
 namespace Microsoft.Bot.Builder.Ai
 {
-
-    //Struct used to store date format and time format for different locales
+    /// <summary>
+    /// DateAndTimeLocaleFormat Class used to store date format and time format
+    /// for different locales.
+    /// </summary> 
     internal class DateAndTimeLocaleFormat
-    { 
+    {
         public string TimeFormat { get; set; }
-        public string DateFormat { get;set; }
+        public string DateFormat { get; set; }
 
     }
 
-    //Struct to store  text and date time object from Microsoft Recognizer recognition result
+    /// <summary>
+    /// TextAndDateTime Class used to store  text and date time object 
+    /// from Microsoft Recognizer recognition result.
+    /// </summary> 
     internal class TextAndDateTime
-    { 
+    {
         public string Text { get; set; }
-        public DateTime DateTimeObject { get; set; }
+        public DateTime dateTime { get; set; }
     }
 
     /// <summary>
-    /// Locale Converter Class used to convert between locales
-    /// in terms of date and time only.
+    /// Locale Converter Class Converts dates and times 
+    /// between different locales.
     /// </summary>
     public class LocaleConverter : ILocaleConverter
-    { 
-        private Dictionary<string, DateAndTimeLocaleFormat> mapLocaleToFunction = new Dictionary<string, DateAndTimeLocaleFormat>();
+    {
 
-        public LocaleConverter()
+        private static readonly Dictionary<string, DateAndTimeLocaleFormat> _mapLocaleToFunction = new Dictionary<string, DateAndTimeLocaleFormat>();
+        private static object _lockMap = new object();
+        private static LocaleConverter _localeConverter;
+        private static readonly Dictionary<string, DateTimeModel> _cacheDateTimeModel = new Dictionary<string, DateTimeModel>();
+        public static LocaleConverter Converter
+        {
+            get
+            {
+
+                if (_localeConverter == null)
```

Comment:
This has concurrency bugs. Consider just creating this init, the same as you do the _cacheDateTimeModel

Reply:
Changed it to ConcurrentDictionary